application/deed/searchdeed/views.py

```diff
@@ -68,27 +68,67 @@ def show_authentication_code_page():
     return render_page
 
 
-def verify_auth_code(auth_code):
+@searchdeed.route('/confirming-mortgage-deed', methods=['POST'])
+def show_confirming_deed_page():
+
+    auth_code = request.form['auth_code']
 
     if auth_code is None or auth_code == '':
         return render_template('authentication-code.html', error=True)
 
-    deed_api_client = getattr(searchdeed, 'deed_api_client')
-    response = deed_api_client.verify_auth_code(str(session.get('deed_token')),
-                                                str(session.get('borrower_token')),
-                                                auth_code)
+    page_render = render_template('confirming-mortgage-deed.html', auth_code=request.form['auth_code'])
+    return page_render
 
-    if response.status_code == status.HTTP_200_OK:
-        return redirect(url_for('searchdeed.show_final_page'), code=307)
-    elif response.status_code == status.HTTP_401_UNAUTHORIZED:
-        return render_template('authentication-code.html', error=True)
-    else:
-        session['code-sent'] = None
-        session['service_timeout_at_send_code'] = None
+
+@searchdeed.route('/verify-auth-code', methods=['POST'])
+def verify_auth_code(auth_code=None):
+    if request.form['auth_code']:
+        auth_code = request.form['auth_code']
+
+    try:
+        deed_api_client = getattr(searchdeed, 'deed_api_client')
+        response = deed_api_client.verify_auth_code(str(session.get('deed_token')),
+                                                    str(session.get('borrower_token')),
+                                                    auth_code)
+
+        if response.status_code == status.HTTP_401_UNAUTHORIZED:
+            return jsonify({'error': True, 'redirect': url_for('authentication-code.html', error=True)})
+        elif response.status_code == status.HTTP_503_SERVICE_UNAVAILABLE or response.status_code \
```

Comment:
again feel a more logical place for the continuation would be after the or instead of mid expression to make the line more readable